### PR TITLE
feat(rhosak): add initial scheduled upgrade template

### PR DESCRIPTION
**Pull request by machi1990**
Also allow all events to be Instant or Daily. The user will have to create a behaviour group and decide how they want to receive them.

Closes MGDSTRM-5089

/cc @davidffrench @akoserwal for internal review 

/cc @gwenneg @josejulio for anything notifications backend related.

### CODE REVIEW

File: backend/src/main/java/com/redhat/cloud/notifications/templates/Rhosak.java [hunk 1]
```diff
@@ -29,7 +33,9 @@ public TemplateInstance getTitle(String eventType, EmailSubscriptionType type) {
         if (eventType.equals(ACTION_REQUIRED)) {
             return Templates.actionRequiredTitle();
         }
-
+        if (eventType.equals(ACTION_REQUIRED)) {
```
Comment (josejulio) [inline backend/src/main/java/com/redhat/cloud/notifications/templates/Rhosak.java:36]:
```suggestion
        if (eventType.equals(SCHEDULED_UPGRADE)) {
```
I think you mean this here

**Comment by josejulio** [comment]
I noticed that there is no change in the aggregator, but several event types were added for daily processing. Is that expected?

File: backend/src/main/resources/templates/Rhosak/scheduledUpgradeBody.html [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        <p>This notification is for your {action.events.size()} OpenShift Streams {#if action.events.size() is 1} instance {#else} instances{/if} below.</p>
+        <ul>
+            {#each action.events}
+            <li style="font-size: 14px;">
+                {it.payload.name}
+            </li>
+            {/each}
+        </ul>
+
+        Your {#if action.events.size() is 1} instance is {#else} instances are{/if} scheduled for upgrade maintenance to {action.context.kafka_version} on {action.context.upgrade_time.toUtcFormat()}
```
Comment (davidffrench) [inline backend/src/main/resources/templates/Rhosak/scheduledUpgradeBody.html:16]:
```suggestion
        Your {#if action.events.size() is 1} instance is {#else} instances are{/if} scheduled for upgrade maintenance to {action.context.kafka_version} Kafka version on {action.context.upgrade_time.toUtcFormat()}
```

Comment (davidffrench) [reply backend/src/main/resources/templates/Rhosak/scheduledUpgradeBody.html:16]: Might be good to call out that it is a Kafka version upgrade.

**Review by davidffrench — COMMENTED** [review]
lgtm, just one comment

File: backend/src/main/java/com/redhat/cloud/notifications/templates/Rhosak.java (hunk 1 above)
Comment (machi1990) [reply backend/src/main/java/com/redhat/cloud/notifications/templates/Rhosak.java]: Good catch. Thanks

File: backend/src/main/resources/templates/Rhosak/scheduledUpgradeBody.html (hunk 2 above)
Comment (machi1990) [reply backend/src/main/resources/templates/Rhosak/scheduledUpgradeBody.html]: Makes total sense, thanks.

**Comment by machi1990** [comment]
> I noticed that there is no change in the aggregator, but several event types were added for daily processing. Is that expected?

Great spot. This is indeed true. For the moment, the daily upgrade aggregator is still empty, and so is the template. When we update the template, we'll work on the aggregator at the same time. This will likely be next week. Let me know if you see any issues with this workflow/division of task.

**Comment by josejulio** [comment]
> Let me know if you see any issues with this workflow/division of task.

No issues :-)

**Review by josejulio — APPROVED** [review]
lgtm

**Comment by josejulio** [comment]
Let us know when you are ready to merge

**Comment by machi1990** [comment]
> Let us know when you are ready to merge

Let's get this in today. Thanks @josejulio @gwenneg

**Review by gwenneg — APPROVED** [review]
LGTM

**Comment by machi1990** [comment]
Thanks

**Pull request merged**